### ISSUE TITLE
Infinite scroll

### ISSUE THREAD

**Issue by oiledCode** [stefanceriu/SCPageViewController#34]
Hello, not really an issue but worth asking anyway. I'm trying to modify your project to support infinite scroll, with no luck so far. So I was wondering if you could point me in the right direction to do it any suggestion is really appreciated.

**Comment by stefanceriu** [comment]
Hey, I reckon you could get this working by listening to the ```didNavigateToPageAtIndex``` delegate callback and either ```deletePagesAtIndexes:``` and ```insertPagesAtIndexes:``` or ```movePageAtIndex:toIndex:``` so you keep the current one centered on screen.
You should able able to achieve everything through the normal public api without having to modify the component.

**Comment by oiledCode** [comment]
Thanks for your answer, But I'm having hard time understanding what you mean. 

Let's say that I have 3 VC, and I present the pageViewController, showing the controller at index 0,the goal for me is to show the controller with index[2] if I scroll to the left or the controller with index[1] if scroll to the right. So is an infinite scroll but is also circular. 

Do you think it will be possible to achieve this without modifying the component? 

Thanks for your time.

**Comment by stefanceriu** [comment]
Okay, going on your example imagine that you configure the pageViewController to start on page 1 with this configuration: ```[C][A][B]``` where position zero hold your first page `A`, position 1 your second page `B` etc.

If the user scrolls left then ```didNavigateToPageAtIndex:``` will be called with page `0` (`C`) and you can move page `B` before `C` by calling ```[self.pageViewController movePageAtIndex:2 toIndex:0 animated:YES completion:nil];``` . If the opposite happens and the user scrolls right then you can `movePageAtIndex:0 toIndex:2`. 

Basically you are trying to keep the pageViewController balanced and centered on page 1 by moving the extremities around.

Oh and if you happen to be building this on top of the demo project please make sure you also update the cache (the `self.viewControllers` array) otherwise you'll see weird results.

**Comment by OhMyApp** [comment]
Hello, I didn't succeed in implementing an infinite scroll.
Could you post some example ?

Or maybe you could helpe me:
I have 3 controllers, `[A] [B] [C]`.
I want an infinite scroll.
I tried to set up this effect when scrolling to the top.
So when I navigate to index 0, I swap controllers to have `[C] [A] [B] `.

Here is the method:

```
func pageViewController(_ pageViewController: SCPageViewController!, didNavigateToPageAt pageIndex: UInt) {
	if pageIndex == 0 {
		guard !isDoingMagic else { return }
		isDoingMagic = true
		pageViewController.movePage(at: 0, to: 2, animated: false) {
			pageViewController.movePage(at: 1, to: 2, animated: false) {
				pageViewController.navigateToPage(at: 1, animated: false) {
					self.isDoingMagic = false
				}
			}
		}
	}
}
```

But it doesn't work.
Do you have an idea ?
